### PR TITLE
Fetch Mime Type From EnumDescriptor

### PR DESCRIPTION
**Pull request by stevenhartley**
In lieu of a switch statement and static defining the mime type, pull the mime type from the proto generated code

#71

### CODE REVIEW

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java [hunk 1]
```diff
@@ -467,24 +469,14 @@ static CloudEvent fromMessageParts(UUri source, UAttributes attributes, UPayload
...
-                return UPayloadFormat.UPAYLOAD_FORMAT_SOMEIP;
-            case "application/x-someip_tlv":
-                return UPayloadFormat.UPAYLOAD_FORMAT_SOMEIP_TLV;
-            case "application/protobuf":
-                return UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF;
-            default:
-                return UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY;
-
+        
+        for (EnumValueDescriptor v : UPayloadFormat.getDescriptor().getValues()) {
```
Comment (tamarafischer) [inline src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java:473]: it is a lot harder to understand what the code is doing, can you add some JavaDoc to explain?

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java [hunk 2]
```diff
@@ -467,24 +469,14 @@ static CloudEvent fromMessageParts(UUri source, UAttributes attributes, UPayload
     static UPayloadFormat getUPayloadFormatFromContentType(String contentType){
         if(contentType == null)
```
Comment (tamarafischer) [inline src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java:470]: want to try using a Stream with a filter.
Nothing wrong with for loops and if statements - just saying.
This method is going to be called a lot - nice if it was efficient.
I take it the unit tests did not change and you ran them all. :)

Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java]: Done please re-review

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java (hunk 1 above)
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java]: Added better English :-)

**Review by neelam-kushwah — APPROVED** [review]

**Review by tamarafischer — APPROVED** [review]
Approving, please check in your **tests** that you do not throw exceptions or barf in any way, you have a strong contract that is someone gives you something you don't know, you return the default. Otherwise we would need to use an Optional or a Result.
You made a change that does not break the client contract, your tests should make sure of that.
Changing code to look nicer or be optimized is great refactoring, as long as you have tests to cover you.

**Pull request merged**